src/test/java/org/sonar/plugins/github/PullRequestIssuePostJobTest.java

```diff
@@ -82,69 +110,23 @@ public void testPullRequestAnalysisNoIssue() {
 
   @Test
   public void testPullRequestAnalysisWithNewIssues() {
-    Issue newIssue = mock(Issue.class);
     DefaultInputFile inputFile1 = new DefaultInputFile("src/Foo.php");
-    when(cache.byKey("foo:src/Foo.php")).thenReturn(inputFile1);
-    when(newIssue.componentKey()).thenReturn("foo:src/Foo.php");
-    when(newIssue.line()).thenReturn(1);
-    when(newIssue.ruleKey()).thenReturn(RuleKey.of("repo", "rule"));
-    when(newIssue.severity()).thenReturn(Severity.BLOCKER);
-    when(newIssue.isNew()).thenReturn(true);
-    when(newIssue.message()).thenReturn("msg1");
+    Issue newIssue = newMockedIssue("foo:src/Foo.php", inputFile1, 1, RuleKey.of("repo", "rule"), Severity.BLOCKER, true, "msg1");
```

Comment:
Depends, do you ever plan to test with different rule keys?
If any non null value does the job then we do not need it as a parameter.